### DISCUSSION TITLE
Get JsRuntime execution output

### DISCUSSION THREAD

**Discussion by gufoe** [denoland/deno#12635] [Q&A] [answered]
Hello, I'd like to reopen the issue #11105 of getting something useful from the script evaluation of deno_code.

So, I'd like to use deno to interpret js at runtime and get the result, for instance:
```rust
let res = runtime.execute_script("<main>", "123");
assert!(res.unwrap() == serde_json::Value::from(123))
```
But I cannot understand how to transform a Global<Value> into something useful.

Is there any way to achieve this?
I know there has been discussion in #11129 about the return value not always being deserializable.
Couldn't a enum be returned?
Something like this:
```rust
enum EvalResult {
  Deserializable(?)
  Other(Global<Value>)
}
```

**Comment by lucacasonato** [comment, marked as answer]
There are two things you can do:

1. You can return a `Global<Value>` into a string using the v8 API:
    ```rs
     let value = isolate.execute_script("<anon>", "'hello' + ' world!'").unwrap();
     let mut scope = isolate.handle_scope();
     let value = value.get(&mut scope);
     let result = value.to_rust_string_lossy(&mut scope);
     println!("{}", result);
    ```
2. You can use `serde_v8` to deserialize a V8 value into any Rust value that implements `serde::Deserialize`:
    ```rs
     let value = isolate.execute_script("<anon>", "'hello' + ' world!'").unwrap();
     let mut scope = isolate.handle_scope();
     let value = value.get(&mut scope);
     let result: String =  serde_v8::from_v8(value, &mut scope);
     println!("{}", result);
    ```

**Reply by gufoe** [reply]
Thank you!
Your scripts are right but don't compile, so here's the correct code:

First we execute the script:
```rust
let global_var = context.execute_script("<anon>", "1+1")
    .expect("Error in script");
let mut scope = context.handle_scope();
```

Then we can either:

1. Convert to rust string (__NOTE:__ you will lose typings, so basically js values "2" and 2 will all be converted to the same string):
```rust
let value = global.open(&mut scope);
let value_str = value.to_rust_string_lossy(&mut scope);
```

2. You can cast to anything that implements `serde::Deserialize`:
```rust
let local_var = deno_core::v8::Local::new(&mut scope, global);
let serde_json = serde_v8::from_v8::<serde_json::Value>(&mut scope, local_var)
    .expect("Unable to deserialize");
```

**Reply by bartlomieju** [reply]
@gufoe thanks for checking out, could you maybe add your example to https://github.com/denoland/deno/tree/main/core/examples ?

**Reply by gufoe** [reply]
Done! I've created #12806
